contrib/autostyle.lua

```diff
@@ -0,0 +1,116 @@
+  images, define a shortcut (lua shortcuts). As I couldn't find an event for
+  when a development is removed, so the autostyle won't be applied again, 
+  this shortcut is also helpful then
+* import your images, or use the shortcut on your already imported images
+* To determine which tag you want, list all tags with exiftool:
+  exiftool -j XE021351.RAF, and find the one you want to use
+  you can check it with 
+  > exiftool -AutoDynamicRange XE021351.RAF
+  Auto Dynamic Range              : 200%
+
+
+LICENSE
+GPLv2
+
+]]
+
+local darktable = require "darktable"
+
+
+-- Forward declare the functions
+local autostyle_apply_one_image,autostyle_apply,exiftool_attribute,capture
+
+-- Tested it with darktable 1.6.1 and darktable git from 2014-01-25
+darktable.configuration.check_version(...,{2,0,2},{2,1,0})
+
+-- Apply the style to an image, if it matches the tag condition
+function autostyle_apply_one_image (image)
+  -- We need the tag, the value and the style_name provided from the configuration string
+  local tag,value,style_name=string.match(darktable.preferences.read("autostyle","exif_tag","string"),"(%g+)%s*=%s*(%g+)%s*=>%s*(%g+)")
+
+  -- check they all exist (correct syntax)
+  assert (tag,"EXIF TAG not found in " .. darktable.preferences.read("autostyle","exif_tag","string"));
+  assert (value,"value to match not found in " .. darktable.preferences.read("autostyle","exif_tag","string"));
+  assert (style_name,"style name not found in " .. darktable.preferences.read("autostyle","exif_tag","string"));
+
+  -- First find the style (we have its name)
+  local styles= darktable.styles
+  local style
+  for _,s in ipairs(styles) do
+	  if s.name == style_name then
+		  style=s
+	  end
+  end
+
+  -- Apply the style to image, if it is tagged
+  local auto_dr_attr= exiftool_attribute(image.path .. '/' .. image.filename,tag);
+  if auto_dr_attr==value then
+--	  darktable.print("Image " .. image.filename .. ": autostyle automatically applied " .. darktable.preferences.read("autostyle","exif_tag","string") )
+	  darktable.styles.apply(style,image)
+--  else
+--	  darktable.print("Image " .. image.filename .. ": autostyle not applied, exif tag " .. darktable.preferences.read("autostyle","exif_tag","string")  .. " not matched: " .. auto_dr_attr)
+  end
+end 
+
+
+function autostyle_apply( shortcut)
+  local images = darktable.gui.action_images
+  for _,image in pairs(images) do
+    autostyle_apply_one_image(image)
+  end
+end
+
+-- Retrieve the attribute through exiftool
+function exiftool_attribute(path,attr)
+  local cmd="exiftool -" .. attr .. " '" ..path.. "'";
+  local exifresult=get_stdout(cmd)
+  local attribute= assert(string.match(exifresult,": (.*)"), "Could not find the attribute " .. attr .. " using the command: <" .. cmd .. ">")
+  return attribute
+end
+
+-- run command and retrieve stdout
+function get_stdout(cmd)
+  -- Open the command, for reading
+  local fd = assert(io.popen(cmd, 'r'))
```

Comment:
please replace io.popen with the DT specific equivalent (yield("READ_FILE") IIRC

the DT version does not block other lua threads while it is running